### PR TITLE
Feature/3DS2 flow api changes

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
- Changes made in response to changes on the API in terms of expected request & response objects
- Temporarily added functions that allows all this new 3DS2 flow code to run against the old 3DS2 flow - using the old actions and endpoints (i.e multiple calls to `/details`) and handling the original request/response objects

## Tested scenarios
Manually tested all 4 flows:
- Frictionless (no challenge)
- Full (fingerprint & challenge)
- Challenge only
- Redirect (fingerprint then redirect)

...against both old and new API versions

Added e2e tests for the old API flows (frictionless and full).

### CODE REVIEW

File: packages/lib/src/components/ThreeDS2/callSubmit3DS2Fingerprint.ts [hunk 1]
```diff
@@ -13,45 +13,39 @@ export default function callSubmit3DS2Fingerprint(state) {
...
 
-        if (!data.action) {
-            console.error('Handled Error::callSubmit3DS2Fingerprint::FAILED:: data=', data);
+        if (!resData.action && !resData.details) {
+            console.error('Handled Error::callSubmit3DS2Fingerprint::FAILED:: resData=', resData);
             return;
         }
 
+        console.log('### callSubmit3DS2Fingerprint::resData:: ', resData);
+
```
Comment (marcperez) [inline packages/lib/src/components/ThreeDS2/callSubmit3DS2Fingerprint.ts:28]: Can we remove this?

File: packages/playground/src/pages/Cards/Cards.js [hunk 2]
```diff
@@ -5,16 +5,15 @@ import { handleSubmit, handleAdditionalDetails, handleError } from '../../handle
...
 
 getPaymentMethods({ amount, shopperLocale }).then(paymentMethodsResponse => {
     window.checkout = new AdyenCheckout({
         amount,
         clientKey: process.env.__CLIENT_KEY__,
         paymentMethodsResponse,
         locale: shopperLocale,
-        countryCode,
         environment: 'test',
+        //        environment: 'http://localhost:8080/checkoutshopper/',
```
Comment (marcperez) [inline packages/playground/src/pages/Cards/Cards.js:16]: Can we remove this line?

File: packages/playground/src/pages/Cards/Cards.js [hunk 3]
```diff
@@ -5,16 +5,15 @@ import { handleSubmit, handleAdditionalDetails, handleError } from '../../handle
...
 import '../../style.scss';
-import { countryCode } from '../../config/commonConfig';
 
 getPaymentMethods({ amount, shopperLocale }).then(paymentMethodsResponse => {
     window.checkout = new AdyenCheckout({
         amount,
         clientKey: process.env.__CLIENT_KEY__,
         paymentMethodsResponse,
         locale: shopperLocale,
-        countryCode,
```
Comment (marcperez) [inline packages/playground/src/pages/Cards/Cards.js:16]: Is this deleted line intentional?

Comment (sponglord) [reply packages/playground/src/pages/Cards/Cards.js:16]: yes. We only added it to make the kcp tests work. Now we have the clientScripts we don't need it anymore

File: packages/lib/src/components/ThreeDS2/callSubmit3DS2Fingerprint.ts (hunk 1 above)
Comment (sponglord) [reply packages/lib/src/components/ThreeDS2/callSubmit3DS2Fingerprint.ts:28]: yes

File: packages/playground/src/pages/Cards/Cards.js (hunk 2 above)
Comment (sponglord) [reply packages/playground/src/pages/Cards/Cards.js:16]: I don't know. We have it in other playground files. It's kind of handy to not have to keep typing it out

Comment (sponglord) [reply packages/playground/src/pages/Cards/Cards.js]: removed

**Review by marcperez — APPROVED** [review]

**Pull request merged**